packages/auth/src/Auth.ts

```diff
@@ -118,16 +118,21 @@ export class Auth {
   }
 
   /**
-   * Provides an Authorization Bearer header token
+   * Provides an Authorization Context which contains a token and a prebuilt header
    * @param tokenUpdateTime time (in seconds) to refresh the token
    */
-  public getHeaderProvider(tokenUpdateTime: number = 10) {
+  public getAuthContextProvider(tokenUpdateTime: number = 10) {
     return () => {
       const tokenUpdate = this.extract().updateToken(tokenUpdateTime) as any;
       // Keycloak doesn't use a proper promise. Instead it uses success/error.
       return new Promise((resolve, reject) => {
         tokenUpdate.success(() => {
-          resolve({ "Authorization": "Bearer " + this.auth.token });
+          resolve({
```

Comment:
World class job! Like that we DRY. Could we made this method to return explicit interface. 
This will be cool for api docs :)